tiledb/sm/cpp_api/query.h

```diff
@@ -231,6 +231,15 @@ class Query {
     return query_layout;
   }
 
+  /** Returns the array of the query. */
```

Comment:
Seth pointed out that the destructor of an `Array` object closes the array, we will not be able to return a new instance of `Array` here.

Instead:
1. Modify this class to contain a `reference_wrapper<const Array> array_` (similar to the existing `reference_wrapper<const Context> ctx_`). 
2. Modify the two constructors that accept a `const Array&` argument and store those in the new `array_`.
3. Modify this API to:
```
const Array& Query::array() {
  return array_;
}
```

Reply:
Cool! I didn't want to become to invasive in the current structure, but this is a good solution. I ll make the change ;)